src/types.ts

```diff
@@ -199,11 +218,17 @@ export interface Composition {
 }
 
 export interface BuildConfig {
```

Comment:
this should largely overlap with https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/dockerode/index.d.ts#L805-L831 which is what downstream projects would probably want this for...